pyfar/dsp/filter/frequency_weighting.py

```diff
@@ -0,0 +1,370 @@
+"""
+Functions and helpers to create frequency weighting filters
+according to IEC 61672-1.
+"""
+import warnings
+from typing import Callable, Literal, Optional
+import numpy as np
+import pyfar as pf
+import pyfar.constants as pfc
+import scipy.signal as sps
+import scipy.optimize as spo
+from pyfar.constants.frequency_weighting import _F_1, _F_2, _F_3, _F_4
+
+
+_A_WEIGHTING_ZEROS = [0, 0, 0, 0]
+_A_WEIGHTING_POLES = [_F_1, _F_1, _F_4, _F_4, _F_2, _F_3]
+_C_WEIGHTING_ZEROS = [0, 0]
+_C_WEIGHTING_POLES = [_F_1, _F_1, _F_4, _F_4]
+
+
+def frequency_weighting_filter(
+        signal,
+        target_weighting: Literal["A", "C"]="A",
+        n_frequencies=100,
+        error_weighting: Optional[Callable[[
+            np.ndarray], np.ndarray]] = None,
+        sampling_rate: Optional[float] = None,
+        **kwargs,
+        ):
+    """
+    Create and apply an A or C weighting filter.
+    The created SOS filter approximates the A or C weighting defined
+    in IEC/DIN-EN 61672-1.
+
+    .. note::
+        This function will run a least-squares algorithm to iteratively
+        approach the target weighting curve. Therefore, it may be much
+        faster to create the filter once and reuse that filter when
+        repeatedly weighting audio data of identical sampling rates.
+
+    .. note::
+        When using default parameters for `n_frequencies` and `error_weighting`
+        and no `kwargs`, the returned filter is compliant with a class 1 sound
+        level meter as described in the standard for the sampling rates
+        48 kHz, 44.1 kHz, 16 kHz as well as these sampling rates multiplied by
+        2, 1/2, 4, 1/4, 8, and 1/8 each.
+        For other arguments or sampling rates, the returned filter may not
+        comply with class 1 requirements, in which case a warning will be
+        printed.
+
+    Parameters
+    ----------
+    signal : Signal, None
+        The signal to be filtered. Pass ``None`` to create the filter without
+        applying it.
+
+    target_weighting: str, optional
+        Specifies which frequency weighting curve to approximate.
+        Must be either "A" or "C". The default is "A".
```

Comment:
Sorry, I missed that before. We have the convention to highlight code in the docstring:
```suggestion
        Must be either ``"A"`` or ``"C"``. The default is ``"A"``.
```
Variables (as in the notes above) should be italic (which they are due to the single ` quotes)